src/wat-writer.cc

```diff
@@ -1382,17 +1381,18 @@ void WatWriter::WriteGlobal(const Global& global) {
   WriteCloseNewline();
 }
 
-void WatWriter::WriteBeginEvent(const Event& event) {
+void WatWriter::WriteEvent(const Event& event) {
   WriteOpenSpace("event");
   WriteNameOrIndex(event.name, event_index_, NextChar::Space);
   WriteInlineExports(ExternalKind::Event, event_index_);
   WriteInlineImport(ExternalKind::Event, event_index_);
-  WriteTypes(event.sig, nullptr);
+  if (event.decl.has_func_type) {
```

Comment:
Don't all events have a type?

Reply:
yes, this flag specifies whether the type is specified like `(event (type $t))`